### PR TITLE
core: Add support for DD_VERSION env

### PR DESCRIPTION
**Pull request by brettlangdon**
Add support for configuring `ddtrace.config.version` from `DD_VERSION`, allow user's to override `span.meta["version"]` via `span.set_tag(VERSION_KEY, "1.2.3")`, automatically set `version` tag on spans by default.

This is broken out from #1211 which also includes changes to inject version into logs.

### CODE REVIEW

File: tests/test_global_config.py [hunk 1]
```diff
@@ -241,3 +242,14 @@ def on_web_request(span):
...
+    def test_dd_version(self):
+        c = Config()
+        assert c.version is None
+
+        with override_env(dict(DD_VERSION="1.2.3")):
+            c = Config()
+            assert c.version == "1.2.3"
+
+            c.version = "4.5.6"
+            assert c.versionn == "4.5.6"
```
Comment (majorgreys) [inline tests/test_global_config.py]: Trailing 'n'?

```suggestion
            assert c.version == "4.5.6"
```

File: tests/test_tracer.py [hunk 2]
```diff
@@ -655,3 +656,24 @@ def task(t, errors):
     assert original_writer._trace_queue.qsize() == 1
     assert t.writer._trace_queue.qsize() == 1
     assert [[span]] == list(t.writer._trace_queue.get())
+
+
+def test_tracer_with_version():
+    t = ddtrace.Tracer()
+
+    # With global `config.version` defined
+    with BaseTracerTestCase.override_global_config(dict(version='1.2.3')):
```
Comment (jdgumz) [inline tests/test_tracer.py]: Do we also have the case covered where someone sets a global tag on the tracer _and_ has `DD_VERSION` available from the env vars?

Comment (brettlangdon) [reply tests/test_tracer.py]: no, we don't, good catch

Comment (brettlangdon) [reply tests/test_tracer.py]: I added a test case here, saying that `DD_VERSION` should take precedence

**Comment by brettlangdon** [comment]
@jd yes, we will want to at some point, but not a hard requirement right now.

File: tests/test_tracer.py [hunk 3]
```diff
@@ -655,3 +656,30 @@ def task(t, errors):
...
+
+        # explicitly set in the span
+        span.set_tag(VERSION_KEY, '1.2.3')
+        assert span.get_tag(VERSION_KEY) == '1.2.3'
+
+    # With global tags set
+    t.set_tags({VERSION_KEY: 'tags.version'})
+    with BaseTracerTestCase.override_global_config(dict(version='config.version')):
+        with t.trace('test.span') as span:
+            assert span.get_tag(VERSION_KEY) == 'config.version'
```
Comment (jdgumz) [inline tests/test_tracer.py]: Might be ridiculous but what if we have:
- `version` set as a global tag
- `DD_VERSION` present
- A span tag for `version`

Comment (jdgumz) [reply tests/test_tracer.py]: Ah, and then there's the permutations involving `service.version`...maybe that's excessive though.

Comment (brettlangdon) [reply tests/test_tracer.py]: `service.version` is a little excessive since it is basically the same as `version` for us.

Comment (brettlangdon) [reply tests/test_tracer.py]: if a customer manually sets `version`, it'll override the global tag or `DD_VERSION`

we just need the default logic.

**Pull request merged**